opensrp-reveal/src/test/java/org/smartregister/reveal/interactor/BaseInteractorTest.java

```diff
@@ -165,6 +189,56 @@ public void testSavePaotForm() throws JSONException {
         assertEquals(structureId, event.getBaseEntityId());
     }
 
+    @Test
+    public void testSaveRegisterStructureForm() throws JSONException {
+        String form = AssetHandler.readFileFromAssetsFolder(org.smartregister.reveal.util.Constants.JsonForm.ADD_STRUCTURE_FORM, context);
+        String planIdentifier = UUID.randomUUID().toString();
+        PreferencesUtil.getInstance().setCurrentPlanId(planIdentifier);
+        JSONObject formObject = new JSONObject(form);
+        String locationId = UUID.randomUUID().toString();
+        formObject.put("entity_id", locationId);
+        JSONObject details = new JSONObject();
+        String taskId = UUID.randomUUID().toString();
+        details.put(LOCATION_PARENT, locationId);
+        formObject.put(DETAILS, details);
+        Whitebox.setInternalState(interactor, "operationalAreaId", locationId);
+        String structureOb = "{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[28.35228319086664,-15.421616685545176,0]},\"properties\":null}";
+        JsonFormUtils.getFieldJSONObject(JsonFormUtils.fields(formObject), "structure").put(VALUE, structureOb);
+        double zoomLevel = 18.2d;
+        JsonFormUtils.getFieldJSONObject(JsonFormUtils.fields(formObject), GeoWidgetFactory.ZOOM_LEVEL).put(VALUE, zoomLevel);
+        Task task = taskUtils.generateTask(context, locationId, locationId,
+                org.smartregister.reveal.util.Constants.BusinessStatus.NOT_VISITED, org.smartregister.reveal.util.Constants.Intervention.IRS, R.string.irs_task_description);
+        when(taskUtils.generateTask(context, locationId, locationId,
+                org.smartregister.reveal.util.Constants.BusinessStatus.NOT_VISITED, org.smartregister.reveal.util.Constants.Intervention.IRS, R.string.irs_task_description))
+                .thenReturn(task);
+        interactor.saveJsonForm(formObject.toString());
+        verify(eventClientRepository, timeout(ASYNC_TIMEOUT)).addEvent(anyString(), eventCaptor.capture());
+        verify(clientProcessor, timeout(ASYNC_TIMEOUT)).processClient(eventClientCaptor.capture(), eq(true));
+        verify(presenter, timeout(ASYNC_TIMEOUT)).onStructureAdded(featureArgumentCaptor.capture(), featureCoordinatesCaptor.capture(), doubleArgumentCaptor.capture());;
```

Comment:
Codacy found an issue: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5712562)